src/producer.ts

```diff
@@ -20,6 +20,7 @@ export class Producer {
     this.validate(options);
     this.queueUrl = options.queueUrl;
     this.batchSize = options.batchSize || 10;
+    options.useQueueUrlAsEndpoint = options.useQueueUrlAsEndpoint ?? true;
```

Comment:
Options should never be overridden for this purpose, how it has been done for other options is how this should be implemented.

Reply:
Also, surely it can just be provided in the options without this line at all?